src/game/Spell.cpp

```diff
@@ -2982,9 +2982,12 @@ void Spell::cast(bool skipCheck)
     // CAST SPELL
     SendSpellCooldown();
 
-    TakePower();
-    TakeReagents();                                         // we must remove reagents before HandleEffects to allow place crafted item in same slot
-    TakeAmmo();
+    if (!m_IsTriggeredSpell)
```

Comment:
Yes, this is legit for PR. In TBC Avenger's Shield was the only spell that should not hit controlled units, this attr has been confirmed on WOTLK branch where many spells got added to this behaviour.